actors/v2/multisig/multisig.go

```diff
@@ -24,7 +24,21 @@ func (*Msig) MsigConstructor(network string, height int64, raw []byte) (map[stri
 		return nil, fmt.Errorf("constructorParams: %s not found", version.String())
 	}
 
-	return parseCBOR(raw, nil, params(), nil)
+	metadata, err := parseCBOR(raw, nil, params(), nil)
+	if err != nil {
+		versions := tools.GetSupportedVersions(network)
+		for _, version := range versions {
+			params, ok := constructorParams[version.String()]
+			if !ok {
+				continue
+			}
+			metadata, err = parseCBOR(raw, nil, params(), nil)
+			if err == nil {
+				break
+			}
+		}
+	}
+	return metadata, err
```

Comment:
This fallback tries to decode with all available versions, right? interesting...